File: Pythonleetcode/Leetcode141.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        if not head:
            return False
        a = head
        exist=[head]
        while True:
            a=a.next
            if a==None:
                return False
            else:
                if a in exist:
                    return True
                exist.append(a)

File: CppPractice/Project1/Project1/Leetcode141.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


struct ListNode {
    int val;
    ListNode* next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode* next) : val(x), next(next) {}
};


class Solution {
public:
    bool hasCycle(ListNode* head) {
        ListNode* cur = head;
        vector<ListNode*> exist;
        while (cur)
        {
            if (find(exist.begin(), exist.end(), cur) == exist.end())
            {
                exist.push_back(cur);
            }
            else
            {
                return true;
            }
            cur = cur->next;


        }

        return false;
    }
};